src/lem.lisp

```diff
@@ -303,3 +303,17 @@ See scripts/build-ncurses.lisp or scripts/build-sdl2.lisp"
                   (insert-string (buffer-point buffer) warning)
                   (insert-character (buffer-point buffer) #\newline))
                 (pop-to-buffer buffer)))))
+
+
+(defun impl-source-buffer-p (buffer)
+  (pathname-match-p (buffer-directory buffer)
+                    (second (assoc "SYS:SRC;**;*.*.*" (logical-pathname-translations "SYS") :test 'equal)))
+  )
```

Comment:
> Still i don't know how to send the eval form to the connected running repl, or maybe there is some better solutions to this.

I have just changed micros and lem to allow this.

https://github.com/lem-project/lem/commit/86d9f36ae1c93d238ec67f2f1671ffa8984d4674
https://github.com/lem-project/micros/commit/25177e04a946365397a0848a6545399860d310d0


The source pattern is now retained for each connection, so that it can be written as follows.
```common-lisp
(dolist (pattern (connection-system-file-patterns (current-connection)))
  (when (pathname-match-p directory pattern)
    (return t)))
```